### ISSUE TITLE
Another typing rule problem

### ISSUE THREAD

**Issue by francoisthire** [Deducteam/Dedukti#18]
```
#NAME test.

B : Type.

T : B.

def A : B -> Type.

def f : A T -> Type.

[] f B --> B.
```

**Comment by rlepigre** [comment]
That is exactly the same "problem" as with [issue 17](https://github.com/Deducteam/Dedukti/issues/17) as far as I'm concerned. The constraint that is used to type-check the LHS is "Type == A T".

**Comment by francoisthire** [comment]
But you are on the kind level, not on type level. There is no way in the
logic defined by the lambda pi modulo calculus that A T can be equal to
Type.

2018-01-15 11:52 GMT+01:00 Rodolphe Lepigre <notifications@github.com>:

**Comment by rlepigre** [comment]
Exactly, so the rule is correct !

**Comment by francoisthire** [comment]
Well, probably the user made a mistake when he entered the rule. Moreover,
I don't see any application where that kind of rule should be used. Hence,
I am not sure that Dedukti should not complain here.



2018-01-15 12:04 GMT+01:00 Rodolphe Lepigre <notifications@github.com>:

**Comment by rlepigre** [comment]
Sure, but I still think we shouldn't care since we will always be able to come up with rules that will never apply (because their constraints cannot be satisfied). Personally, I think that we should either reject all such rule, or none of them. And it's probably difficult to reject all of them.

**Comment by francoisthire** [comment]
For the moment, I think we should reject all of them because in the paper
"expressing" : http://www.lsv.fr/~dowek/Publi/expressing.pdf (page 8),
typing the rewrite rule l --> r implies that the m.g.u. \sigma, l\sigma,
and r\sigma have a type T. In these examples, such T does not exist. If we
change the notion of typing, I would like to have a soundness proof first
(or a really good argument).

2018-01-15 13:53 GMT+01:00 Rodolphe Lepigre <notifications@github.com>:

**Comment by fblanqui** [comment]
What is the definition of mgu here?

**Comment by rafoo** [comment]
> even in the first-order case, if rewriting rules are taken into account

Are you requesting a feature? What is the problem with working with Miller mgus?

**Comment by fblanqui** [comment]
I just want to underline the fact that 1) the notion of mgu that is used 
is not defined, 2) why considering mgu modulo beta and eta only in a 
framework that claims to use rewriting?, 3) why would it be sufficient 
to consider a mgu modulo beta and eta only to check subject reduction?


Le 26/01/2018 à 08:14, Raphaël Cauderlier a écrit :

**Comment by francoisthire** [comment]
Fixed in #108 .
